ironfish/src/workerPool/serializableMessage.ts

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import bufio from 'bufio'
+import { Serializable } from '../common/serializable'
+
+export enum WorkerMessageType {
+  BoxMessage = 'boxMessage',
+  CreateMinersFee = 'createMinersFee',
+  CreateTransaction = 'createTransaction',
+  GetUnspentNotes = 'getUnspentNotes',
+  MineHeader = 'mineHeader',
+  Sleep = 'sleep',
+  SubmitTelemetry = 'submitTelemetry',
+  TransactionFee = 'transactionFee',
+  UnboxMessage = 'unboxMessage',
+  VerifyTransaction = 'verifyTransaction',
+}
+
+export abstract class SerializableWorkerMessage implements Serializable {
+  constructor(id: number, type: WorkerMessageType) {
+    this.id = id
+    this.type = type
+  }
+
+  id: number
+
+  type: WorkerMessageType
+
+  abstract serialize(bw: bufio.BufferWriter): Buffer
```

Comment:
I wonder if we should just return `BufferWriter` here instead. It feels a bit odd to take a writer and then render it per serialization. It might be cleaner to either (1) return the updated writer instance (`serialize: (BufferWriter) -> BufferWriter`) or (2) take no writer and just return a buffer which the caller can use (`serialize: () -> Buffer`). 

If doing the first option, we will need to update [this line](https://github.com/iron-fish/ironfish/pull/1093/files#diff-4c76277f743b576804dcad6904503ff146f239d76860b1b6d6c1afab89fa6729R41) to `this.serialize(bw)`.

Reply:
I think that the second option makes more sense as I would expect `serialize()` to return a Buffer.

Reply:
Cool that sounds good 👍